x-pack/solutions/search/plugins/search_homepage/public/hooks/use_search_home_page_redirect.tsx

```diff
@@ -35,27 +41,23 @@ export const useSearchHomePageRedirect = () => {
     }
 
     if (userPrivileges?.privileges?.canManageIndex === false) {
-      setHasDoneRedirect(true);
+      setRedirectChecked(true);
       return;
     }
 
-    if (!indicesStatus) {
+    if (!indicesStatus || error) {
       return;
     }
     if (indicesStatus.indexNames.length === 0) {
-      application.navigateToApp('elasticsearchStart');
-      setHasDoneRedirect(true);
```

Comment:
This was causing an issue:
`navigateToApp` doesn't immediately mount the new route's component. `setHasDoneRedirect` gets enqueued in the call stack and triggers a re-render of the consumer component (`search_homepage.tsx`). The homepage saw that `isLoading` is done and `hasDoneRedirect` is true, but the homepage is still mounted. So homepage goes ahead and renders the new CTA variant and fires off a tracking event that the variant was loaded.
Then the `/start` component is mounted and then the homepage is unmounted.

We need to leave `isLoading` as true from the consumer's point of view until the redirect has happened. `redirectChecked` essentially acts as a lock that lets the redirect finish. There may be a better way if we dig into `navigateToApp` but this works for now